### PR TITLE
Remove check for ssl_enabled for ConfigBuilder

### LINKED ISSUE #22252: Error running test-wpt on Windows
```
[vcvarsall.bat] Environment initialized for: 'x64'
 0:03.09 INFO Updating test manifest D:\Workspace\servo\tests\wpt\metadata\MANIFEST.json
 8:58.48 INFO Updating test manifest D:\Workspace\servo\tests\wpt\mozilla\meta\MANIFEST.json
 9:23.93 INFO Updating test manifest D:\Workspace\servo\tests\wpt\webgl\meta\MANIFEST.json
10:30.65 INFO Using 8 client processes
10:33.95 SUITE_START: web-platform-test - running 1 tests
10:33.95 INFO Running reftest tests
10:33.99 INFO Closing logging queue
10:34.00 INFO queue closed
Error running mach:

    ['test-wpt', 'tests\\wpt\\web-platform-tests\\html\\dom\\documents\\dom-tree-accessors\\document.getElementsByName\\document.getElementsByName-liveness.html']

The error occurred in code that was called by the mach command. This is either
a bug in the called code itself or in the way that mach is calling it.

You should consider filing a bug for this issue.

If filing a bug, please include the full output of mach, including this error
message.

The details of the failure are as follows:

TypeError: 'NoneType' object has no attribute '__getitem__'

  File "D:\Workspace\servo\python\servo\testing_commands.py", line 375, in test_wpt
    ret = self.run_test_list_or_dispatch(kwargs["test_list"], "wpt", self._test_wpt, **kwargs)
  File "D:\Workspace\servo\python\servo\testing_commands.py", line 415, in run_test_list_or_dispatch
    return correct_function(**kwargs)
  File "D:\Workspace\servo\python\servo\testing_commands.py", line 401, in _test_wpt
    return self.wptrunner(run_file, **kwargs)
  File "D:\Workspace\servo\python\servo\testing_commands.py", line 447, in wptrunner
    return run_globals["run_tests"](**kwargs)
  File "D:\Workspace\servo\tests\wpt\run.py", line 47, in run_tests
    success = wptrunner.run_tests(**kwargs)
  File "D:\Workspace\servo\tests\wpt\web-platform-tests\tools\wptrunner\wptrunner\wptrunner.py", line 226, in run_tests
    **kwargs)
  File "D:\Workspace\servo\tests\wpt\web-platform-tests\tools\wptrunner\wptrunner\browsers\servo.py", line 36, in browser_kwargs
    "ca_certificate_path": config.ssl_config["ca_cert_path"],
```

### PR DESCRIPTION
**Pull request by KiChjang**
Fixes #22252. This is a rebased version of #22253.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22535)

### CODE REVIEW

**Comment by KiChjang** [comment]
Oh crud, something's wrong with the WPT sync. cc @jdm

**Comment by KiChjang** [comment]
r? @jdm 

This should fix the problems I'm encountering on Windows. Essentially, the SSL config getter should ALWAYS return a record, regardless of whether or not SSL was enabled.

**Comment by KiChjang** [comment]
Should probably cc @jgraham as well.

**Comment by jdm** [comment]
@KiChjang What is the effect of these changes on your local build?

**Comment by jgraham** [comment]
I think this is probably OK.

**Comment by jdm** [comment]
Actually, I think I see how the SSL stuff works and why it's not working on Windows. Rather than making this change, I propose the following:
* in the `build_env` method in the msvc branch, add the openssl package's binary directory to the PATH

Since wptcommandline's `--openssl-binary` defaults to `openssl`, this will allow windows and non-windows platforms to take the same code path, and should allow running tests on Windows.

**Comment by jdm** [comment]
The other reasonable solution here would be modifying our WPT initialization to use the pregenerated certs from tests/wpt/web-platform-tests/tools/certs by setting the `--ca-cert-path`, `--host-cert-path`, and `--host-key-path` arguments appropriately.

File: tests/wpt/web-platform-tests/tools/wptserve/wptserve/config.py [hunk 1]
```diff
@@ -316,10 +316,9 @@ def _get_ssl_config(self, data):
         kwargs = data["ssl"].get(ssl_type, {})
         self._ssl_env = ssl_cls(self.logger, **kwargs)
         self._ssl_env.__enter__()
-        if self._ssl_env.ssl_enabled:
```
Comment (KiChjang) [inline tests/wpt/web-platform-tests/tools/wptserve/wptserve/config.py]: @jdm I think the problem is that the python script keeps thinking that I am in a `NoSSLEnvironment` even after when I add the openssl binary directory to PATH.

**Comment by KiChjang** [comment]
So, I found out that the `--ssl-type` command line argument defaults to `none`, which causes the WPT runner to not use any SSL config at all, which means even if I did set the openssl binary path correctly for windows, it still wouldn't use any SSL config at all. @jdm Should we just default `ssl_type` to `openssl` altogether? Or would `pregenerated` make more sense here?

**Comment by KiChjang** [comment]
I'm also seeing that the code behaves extremely weirdly where defaults are applied _after_ the CLI kwargs are parsed, and not during parsing. This leads to surprising behavior where if the `ssl_type` is changed in the middle of some code, it would somehow Just Work and use the correct openssl binary/certs, even though the corresponding CLI args aren't passed into wptrunner.

**Comment by jdm** [comment]
If I'm correct, your previous two comments refer to these bits of code:
* [ssl_type defaults to none](https://github.com/web-platform-tests/wpt/blob/master/tools/wptrunner/wptrunner/wptcommandline.py#L236-L238)
* [ssl_type is automatically changed to a non-none value, sometimes](https://github.com/web-platform-tests/wpt/blob/81bfcdbf4e6c2b6fba3d18befc64538a01f4a2cb/tools/wptrunner/wptrunner/wptcommandline.py#L496-L502)

This is why I suggested adding the openssl binary to the path on Windows. Really, though, everything will be easier if we set up our ca_cert_path/host_cert_path/host_key_path defaults to point at the pregenerated certs on all platforms. It should be possible to do that in [set_defaults](https://github.com/servo/servo/blob/681c0ebf4b5a4b5abf0843986036e3594cc193b8/tests/wpt/run.py#L51-L79), which calls check_args.

**Comment by jdm** [comment]
Closing in favour of the description in #23133.

**Pull request closed without merging**